conans/client/build/cmake_toolchain_build_helper.py

```diff
@@ -47,198 +43,103 @@ class CMakeToolchainBuildHelper(BaseCMakeBuildHelper):
-                             " one declared in the settings")
-
-    @property
-    def in_local_cache(self):
-        try:
-            in_local_cache = self._definitions[cmake_in_local_cache_var_name]
-            return get_bool_from_text(str(in_local_cache))
-        except KeyError:
-            return False
-
-    @property
-    def flags(self):
-        return defs_to_string(self._definitions)
-
-    @property
-    def command_line(self):
-        return self.flags
-
-    @property
-    def build_config(self):
-        """ cmake --build tool have a --config option for Multi-configuration IDEs
-        """
-        if self._build_type and self.is_multi_configuration:
-            return "--config %s" % self._build_type
-        return ""
-
-    @property
-    def definitions(self):
-        raise ConanException("Cannot add 'definitions' to CMake build helper when using a"
-                             " toolchain. All the definitions should be added to the toolchain")
-
-    @property
-    def runtime(self):
-        raise ConanException("Runtime is assigned by the toolchain and it is not known in the"
-                             " CMakeToolchainBuildHelper helper")
-
-    @property
-    def is_multi_configuration(self):
-        return self._is_multiconfiguration
-
-    @property
-    def verbose(self):
-        try:
-            verbose = self._definitions[verbose_definition_name]
-            return get_bool_from_text(str(verbose))
-        except KeyError:
-            return False
-
-    @verbose.setter
-    def verbose(self, value):
-        self._definitions.update(verbose_definition(value))
-
-    def _get_dirs(self, source_folder, build_folder, cache_build_folder):
-
-        def get_dir(folder, origin):
-            if folder:
-                return os.path.join(origin, folder)
-            return origin
-
-        build_ret = get_dir(build_folder, self._conanfile.build_folder)
-        source_ret = get_dir(source_folder, self._conanfile.source_folder)
-
-        if self._conanfile.in_local_cache and cache_build_folder:
-            build_ret = get_dir(cache_build_folder, self._conanfile.build_folder)
-
-        return source_ret, build_ret
-
-    def configure(self, args=None, defs=None, source_dir=None, build_dir=None,
-                  source_folder=None, build_folder=None, cache_build_folder=None,
-                  pkg_config_paths=None):
-        assert args is None, "All the 'args' should be provided in the toolchain"
-        assert defs is None, "All the 'defs' should be provided in the toolchain"
-        assert source_dir is None, "'source_dir' is deprecated. Use 'source_folder'"
-        assert build_dir is None, "'build_dir' is deprecated. Use 'source_folder'"
-        assert pkg_config_paths is None, "'pkg_config_paths' should be provided in the toolchain"  # TODO: environment?
-
+    def configure(self, source_folder=None, build_folder=None, build_type=None):
```

Comment:
If we can assign the `build_folder` in the constructor, why do we want it here? I would remove it from the `configure` and the `build` methods

Reply:
Because you might want to use different build folders, for different configurations in mult-config. If we are definitely removing the multi-config packages, then this could be simplified.

Reply:
If it is a multi-config generator, then you can have the same build_folder for all the build_types, that's why it is multi-config... if the generator is not multi-config, then you cannot reuse the same toolchain file for different build_types. Is it right?